pkgs/by-name/bs/bsync/package.nix

```
@@ -0,0 +1,31 @@
+{ lib
+, fetchFromGitHub
+, stdenv
+}:
+
+stdenv.mkDerivation {
+  pname = "bsync";
+  version = "unstable-2023-12-21";
+
+  src = fetchFromGitHub {
+    owner  = "dooblem";
+    repo   = "bsync";
+    rev    = "25f77730750720ad68b0ab2773e79d9ca98c3647";
+    hash   = "sha256-k25MjLis0/dp1TTS4aFeJZq/c0T01LmNcWtC+dw/kKY=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm555 bsync -t $out/bin
+    runHook postInstall
+  '';
+
```

I had a look at upstream. The program seems to have a bunch of runtime dependencies, that we'll need to add to the `$PATH` of the program. These ones were the ones I found, but there might be something I've overlooked. You'll have to add the packages referenced to the arguments up top as well.

```suggestion

nativeBuildInputs = [ makeWrapper ];
buildInputs = [ python3 ];

fixupPhase = ''
  runHook preFixup

  patchShebangs $out/bin/bsync
  wrapProgram $out/bin/bsync \
    --prefix PATH ":" ${lib.makeLibraryPath [ openssh rsync findutils ]}

  runHook postFixup
'';

```

---

Yes, I think you should be able open a pure nix-shell with the package like this:

```
nix-shell --pure -A bsync
```